x-pack/plugins/task_manager/server/plugin.ts

```diff
@@ -35,7 +36,9 @@ export type TaskManagerStartContract = Pick<
   TaskScheduling,
   'schedule' | 'runNow' | 'ensureScheduled'
 > &
-  Pick<TaskStore, 'fetch' | 'get' | 'remove'>;
+  Pick<TaskStore, 'fetch' | 'get' | 'remove'> & {
+    deleteTaskIfItExists: (id: string) => Promise<void>;
+  };
```

Comment:
Since `deleteTaskIfItExists` is basically `remove` with a catch for the error, can we do this instace?

```suggestion
  Pick<TaskStore, 'fetch' | 'get' | 'remove'> & {
    deleteTaskIfItExists: TaskStore['remove'];
  };
```

This makes it clear these two are aligned in typing